src/modules/launcher/Wox.Infrastructure/StringMatcher.cs

```diff
@@ -147,14 +156,30 @@ public MatchResult FuzzyMatch(string query, string stringToCompare, MatchOption
             // proceed to calculate score if every char or substring without whitespaces matched
             if (allQuerySubstringsMatched)
             {
-                var score = CalculateSearchScore(query, stringToCompare, firstMatchIndex, lastMatchIndex - firstMatchIndex, allSubstringsContainedInCompareString);
+                var nearestSpaceIndex = CalculateClosestSpaceIndex(spaceIndices, firstMatchIndex);
+                var score = CalculateSearchScore(query, stringToCompare, firstMatchIndex - nearestSpaceIndex - 1, lastMatchIndex - firstMatchIndex, allSubstringsContainedInCompareString);
 
                 return new MatchResult(true, UserSettingSearchPrecision, indexList, score);
             }
 
             return new MatchResult (false, UserSettingSearchPrecision);
         }
 
+        // To get the index of the closest space which preceeds the first matching index
+        private int CalculateClosestSpaceIndex(List<int> spaceIndices, int firstMatchIndex)
```

Comment:
I think we can track this value in the string matching `for` loop itself. Any specific reason to calculate it separately ?

Reply:
So the reason I didn't do it in the above nested for loop is for two reasons -
1. The above for loop is calculated for each space separated query corresponding to each result. So, if we do this evaluation in the loop we'd be performing the operation as many times as the number of space separated queries instead of doing it once, as done in this case.
2. Also, if we do it in the loop, we'd be forced to keep updating it everytime the firstMatchIndex is changed, we'd have to calculate the closest preceeding space rather than doing it once right before the score is calculated. 
I called a different function just to keep the code cleaner.
Pls let me know if you think there is a more efficient way of doing it. Thanks!

Reply:
Sound good ! `firstMatchIndex` reduction requires keeping track of spaces in a list. Thanks for the clarification!